dask/array/slicing.py

```diff
@@ -11,27 +12,47 @@
 from ..compatibility import long
 
 
-def sanitize_index_elements(ind):
+def sanitize_index(ind):
     """ Sanitize the elements for indexing along one axis
 
-    >>> sanitize_index_elements([2, 3, 5])
+    >>> sanitize_index([2, 3, 5])
     [2, 3, 5]
-    >>> sanitize_index_elements([True, False, True, False])
+    >>> sanitize_index([True, False, True, False])
     [0, 2]
-    >>> sanitize_index_elements(np.array([1, 2, 3]))
+    >>> sanitize_index(np.array([1, 2, 3]))
     [1, 2, 3]
-    >>> sanitize_index_elements(np.array([False, True, True]))
+    >>> sanitize_index(np.array([False, True, True]))
     [1, 2]
-    >>> type(sanitize_index_elements(np.int32(0)))
+    >>> type(sanitize_index(np.int32(0)))
     <type 'int'>
+    >>> sanitize_index(1.0)
+    1
+    >>> sanitize_index(0.5)
+    Traceback (most recent call last):
+    ...
+    IndexError: Bad index.  Must be integer-like: 0.5
     """
-    if isinstance(ind, np.integer):
-        ind = int(ind)
+    if isinstance(ind, Number):
+        ind2 = int(ind)
+        if ind2 != ind:
```

Comment:
Arguably, you should use `operator.index` instead -- it exists for exactly this purpose.